cala-ledger/src/velocity/balance/mod.rs

```diff
@@ -72,14 +78,26 @@ impl VelocityBalances {
             Vec<(&AccountVelocityLimit, &EntryValues)>,
         > = HashMap::new();
         for entry in entries {
-            let (_, controls) = match controls.get(&entry.account_id) {
-                Some(control) => control,
-                None => continue,
-            };
-            for control in controls.iter() {
-                let ctx = context.context_for_entry(entry);
-
-                if control.needs_enforcement(&ctx)? {
+            let ctx = context.context_for_entry(entry);
+
+            let account_id = entry.account_id;
+            let mut ids = vec![account_id];
+            ids.extend(
+                account_set_mappings
+                    .get(&account_id)
+                    .unwrap_or(&vec![])
+                    .iter()
+                    .map(AccountId::from)
+                    .collect::<Vec<_>>(),
+            );
+            for account_id in ids {
+                let Some((_, controls)) = controls.get(&account_id) else {
+                    continue;
+                };
+                for control in controls.iter() {
+                    if !control.needs_enforcement(&ctx)? {
```

Comment:
This breaks since context is unable to fetch account values for account set id

Reply:
there are only account values cached in the context if there is actually a control.
Also `let ctx = context.context_for_entry(entry);` will have to receive the `account_id` as well since we may be enforcing on an account that is not the `entry.account_id`.

Reply:
I have fixed this instance [here](https://github.com/GaloyMoney/cala/pull/502/commits/08d1f672df32a8f076daf22f4ae07da986d036a4) but the same thing needs to be applied at the snapshot creation loop.